ui/desktop/tests/e2e/authentication.spec.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+/* eslint-disable no-undef */
+const { test } = require('./fixtures/baseTest');
+const LoginPage = require('./pages/loginPage');
+const { expect } = require('@playwright/test');
+
+/* eslint-disable no-undef */
+test.describe('user/password authentication tests', async () => {
+  test('Authenticates and signs out', async ({
+    electronPage,
+    clusterUrl,
+    username,
+    password,
+  }) => {
+    const loginPage = new LoginPage(electronPage);
+    await loginPage.setClusterUrl(clusterUrl);
+    await loginPage.loginWithPassword(username, password);
+
+    await expect(electronPage).toHaveURL(
+      /.*\/scopes\/global\/projects\/targets$/,
+    );
+
+    await electronPage
+      .locator('details')
+      .filter({ hasText: 'Sign Out' })
+      .click();
+
+    await electronPage.getByRole('button', { name: 'Sign Out' }).click();
+    await expect(electronPage).toHaveURL(/.*\/scopes\/global\/authenticate\//);
+  });
+
+  test('Fails to authenticate with wrong password', async ({
+    electronPage,
+    clusterUrl,
+    username,
+  }) => {
+    const loginPage = new LoginPage(electronPage);
+    await loginPage.setClusterUrl(clusterUrl);
+    await loginPage.loginWithPassword(username, 'wrong password!!');
+
+    await expect(electronPage).toHaveURL(/.*\/scopes\/global\/authenticate\//);
+    await expect(electronPage.getByText('Authentication Failed')).toBeVisible();
+  });
+});
+
+test.describe('LDAP authentication tests', async () => {
```

Comment:
My initial thought was to avoid using `boundary dev` and have a full dockerized setup to truly be end-to-end. However, this would mean we would need to do some additional setup on boundary's end to create an LDAP auth method and the associated resources. This is where I thought some shared methods might make sense (perhaps some methods that use HTTP endpoints). 

Thoughts?

Reply:
Yeah agreed. To be clear, this is just the start to get things setup and having the tests passing again. I think ideally in the long run we'd want to use a real boundary setup and probably enos to get an accurate end to end tests.

But in the meantime I was just using boundary dev and am likely to use some of the admin UI e2e cli methods to set some things up so we can get some tests working quicker